File: Problems/Algorithms/342. Power of Four/power_of_four.rs
impl Solution {
    pub fn is_power_of_four(n: i32) -> bool {
        if n <= 0 {
            return false;
        }
        if n == 1 {
            return true;
        }
        match n % 4 {
            0 => Self::is_power_of_four(n / 4),
            _ => false,
        }
    }
}

File: Problems/Algorithms/342. Power of Four/power-of-four.cpp
#include <cmath>

using namespace std;

class Solution {
public:
    bool isPowerOfFour(int n) {
        if (n == 1) {
            return true;
        }

        if (n <= 0) {
            return false;
        }

        double log4 = log(n) / log(4);
        return (log4 == (int)log4);
    }
};
